frontend/public/components/edit-yaml.jsx

```diff
@@ -278,47 +305,69 @@ export const EditYAML_ = connect(stateToProps)(
           });
         }
 
-        save() {
-          const { onSave, t } = this.props;
-          const { owner } = this.state;
-          let obj;
+        performDryRun(objs) {
+          const { models } = this.props;
+          let dryRunResources = [];
+          this.setState({ success: null, errors: null, sending: true }, () => {
+            dryRunResources = createResources(objs, models, true);
+            //catch these individually so we can report out all errors
+            dryRunResources.forEach((request, i) =>
+              request.catch((error) => this.handleErrors(objs[i], error?.message)),
+            );
+            Promise.all(dryRunResources)
+              .then(() => {
+                this.setState({
+                  errors: null,
+                  sending: false,
+                  resourceObjects: objs,
+                });
+                this.showResults(true);
+              })
+              .catch(() => {
+                //catch this error but do nothing since we show individual errors above
+              });
+          });
+        }
 
-          if (onSave) {
-            onSave(this.getEditor().getValue());
-            return;
-          }
+        showResults(value) {
+          this.props.clearFileUpload();
+          this.setState({ displayResults: value });
+        }
 
-          try {
-            obj = safeLoad(this.getEditor().getValue());
-          } catch (e) {
-            this.handleError(t('public~Error parsing YAML: {{e}}', { e }));
-            return;
+        onRetry(retryObjs) {
+          this.showResults(false);
+          if (retryObjs) {
+            const yamlTextData = [];
+            retryObjs.forEach((yamlObj) => {
+              yamlTextData.push(this.convertObjToYAMLString(yamlObj).trim());
+            });
```

Comment:
Better to use `Array.map` here. Also naming nit: they aren't "YAML objects," they're objects. `yamlObj` makes it sound like it's already the YAML string. I would call them `yamlDocuments` as well.

```suggestion
            const yamlDocuments = retryObjs.map((obj) => {
              return this.convertObjToYAMLString(obj).trim();
            });
```

Reply:
sounds good...or even:
`const yamlDocuments = retryObjs.map((obj) => this.convertObjToYAMLString(obj).trim());`